web/src/beta/features/Editor/Map/DataSourceLayerCreator/WmsTiles/index.tsx

```diff
@@ -61,62 +61,70 @@ const WmsTiles: FC<DataProps> = ({ sceneId, onSubmit, onClose }) => {
+            value={wmsUrlValue}
+            onChange={value => handleOnChange(value, "wmsUrl")}
           />
-        </InputGroup>
-        <InputGroup
-          label={t("Choose layer to add")}
-          description={t("Layer of the data source you want to add.")}>
+        </InputsWrapper>
+      </InputGroup>
+      <InputGroup label={t("Choose layer to add")}>
+        <LayerNameListWrapper>
           {layers.map((layer: string, index: number) => (
             <LayerWrapper key={index}>
-              <Input type="text" value={`${layer}`} disabled={true} />
-              <DeleteLayerIcon icon="bin" size={16} onClick={() => handleDeleteLayer(index)} />
+              <TextInput value={`${layer}`} extendWidth />
+              <Button
+                icon="close"
+                iconButton
+                shadow={false}
+                appearance="simple"
+                size="small"
+                iconColor={theme.content.main}
+                onClick={() => handleLayerNameDelete(index)}
+              />
             </LayerWrapper>
           ))}
-          {(!layers.length || layerInput) && (
+          {(!layers.length || isLayerName) && (
             <LayerWrapper>
-              <Input
-                type="text"
+              <TextInput
                 placeholder={t("layer name")}
-                value={layerValue}
-                onChange={e => setLayerValue(e.target.value)}
-                onKeyDown={handleAddLayer}
+                value={layerNameValue}
+                extendWidth
+                onBlur={handleOnBlur}
+                onChange={value => setLayerNameValue(value)}
+              />
+              <Button
+                icon="close"
+                iconButton
+                shadow={false}
+                size="small"
+                iconColor={theme.content.weak}
+                appearance="simple"
+                disabled
               />
-              <DeleteLayerIcon disabled={true} icon="bin" size={16} />
             </LayerWrapper>
           )}
+        </LayerNameListWrapper>
 
-          <AddLayerWrapper>
-            <Button
-              icon="plus"
-              text={t("Layer")}
-              buttonType="primary"
-              size="small"
-              onClick={handleLayerInput}
-              disabled={!layerValue && !layers.length}
-            />
-          </AddLayerWrapper>
-        </InputGroup>
-      </AssetWrapper>
+        <Button
+          icon="plus"
+          title={t("Layer name")}
+          size="small"
+          onClick={handleLayerNameButtonClick}
+        />
+      </InputGroup>
       <SubmitWrapper>
         <Button
-          text={t("Add to Layer")}
-          buttonType="primary"
-          size="medium"
+          title={t("Add to Layer")}
+          appearance="primary"
           onClick={handleSubmit}
-          disabled={!urlValue}
+          disabled={!wmsUrlValue.endsWith(".wms")}
```

Comment:
Please double check with Red about the validation. Maybe we don't need to check the extention type.